modules/apps/foundation/roles/roles-admin-web/src/main/java/com/liferay/roles/admin/web/internal/portlet/RolesAdminPortlet.java

```diff
@@ -282,6 +283,11 @@ public void updateActions(
 
 		long roleId = ParamUtil.getLong(actionRequest, "roleId");
 
+		long groupId = themeDisplay.getScopeGroupId();
+
+		_permissionService.checkPermission(
+			groupId, Role.class.getName(), roleId);
```

Comment:
Actions should never check permissions. If permission checks are necessary then you need to create the methods in the RemoteLocalService, invoke those from the action and add the permission checks there.